src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/ExecutableStatementCountCheck.java

```diff
@@ -219,33 +226,16 @@ private void leaveMemberDef(DetailAST ast) {
      * @param ast the token representing the statement list.
      */
     private void visitSlist(DetailAST ast) {
-        if (context.getAST() != null) {
-            // find member AST for the statement list
-            final DetailAST contextAST = context.getAST();
-            DetailAST parent = ast.getParent();
-            int type = parent.getType();
-            while (type != TokenTypes.METHOD_DEF
-                && !isConstructorOrInit(type)) {
-                parent = parent.getParent();
-                type = parent.getType();
-            }
-            if (parent == contextAST) {
-                context.addCount(ast.getChildCount() / 2);
-            }
+        final DetailAST contextAST = context.getAST();
+        DetailAST parent = ast.getParent();
+        while (!TokenUtil.isOfType(parent, TokenTypes.METHOD_DEF,
```

Comment:
here is idea: 
lets move this condition check to separate private method and reuse newly created method in `leaveToken` and `visitToken` methods

Reply:
Done. Is it fine now?